From 05414fd66225218324878fb446cb9b74d524a4ad Mon Sep 17 00:00:00 2001
From: Dennis Bonke <admin@dennisbonke.com>
Date: Sun, 16 Apr 2023 20:36:43 +0200
Subject: [PATCH 3/3] Disable futex for managarm

With the new linux-headers, we now have a <linux/futex.h> present. mesa
seems to guard using the linux syscalls for futex support only behind
the presence of this header, and not by OS - which is incorrect, as the
code assumes Linux due to raw syscalls. Here, we change it to force both
the header to be present and to be linux to use that path, and use the
fallback for managarm (as has been the case previously)

Signed-off-by: Dennis Bonke <admin@dennisbonke.com>
---
 src/util/futex.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/util/futex.h b/src/util/futex.h
index c397507..995c1e2 100644
--- a/src/util/futex.h
+++ b/src/util/futex.h
@@ -24,7 +24,7 @@
 #ifndef UTIL_FUTEX_H
 #define UTIL_FUTEX_H
 
-#if defined(HAVE_LINUX_FUTEX_H)
+#if defined(HAVE_LINUX_FUTEX_H) && defined(__linux__)
 #define UTIL_FUTEX_SUPPORTED 1
 #elif defined(__FreeBSD__)
 #define UTIL_FUTEX_SUPPORTED 1
-- 
2.42.0

